### PR TITLE
MCR-3053 fix instantiation of MCROAISolrSetHandler

### PR DESCRIPTION
**Pull request by kkrebs**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3053).

comes from https://github.com/MyCoRe-Org/mycore/commit/ec5de0a8ec4faf998b7e28ad2ddc7d4f6b7ef36a#diff-63774fe31cf1f8d417b64d01680d7715211199cbb259f3e5699188a4c040b81e

### CODE REVIEW

File: mycore-oai/src/main/java/org/mycore/oai/set/MCROAISolrSetConfiguration.java [hunk 1]
```diff
@@ -44,8 +44,7 @@ public MCROAISolrSetConfiguration(String configPrefix, String setId) {
         String setConfigPrefix = configPrefix + SETS_PREFIX + setId;
         MCROAISolrSetHandler handler = MCRConfiguration2.getInstanceOf(
                 MCROAISolrSetHandler.class, setConfigPrefix + ".Handler")
-            .orElseGet(() -> MCRConfiguration2.getInstanceOfOrThrow(
-                MCROAISolrSetHandler.class, getFallbackHandler(configPrefix, setId)));
+            .orElseGet(() -> MCRConfiguration2.instantiateClass(getFallbackHandler(configPrefix, setId)));
```
Comment (toKrause) [inline mycore-oai/src/main/java/org/mycore/oai/set/MCROAISolrSetConfiguration.java]: The fallback handler might be be `null`, support for which was unintentionally dropped in MCR-3053.

Maybe:

```java
MCROAISolrSetHandler handler = MCRConfiguration2
    .getInstanceOf(MCROAISolrSetHandler.class, setConfigPrefix + ".Handler")
    .orElseGet(() -> {
        String fallbackHandler = getFallbackHandler(configPrefix, setId);
        if (fallbackHandler == null) {
            throw new MCRConfigurationException("Configuration property " +  setConfigPrefix 
                + ".Handler is not set and no fallback handler could be determined");
        }
        return MCRConfiguration2.instantiateClass(fallbackHandler);
    });
```

**Review by toKrause — CHANGES_REQUESTED** [review]

File: mycore-oai/src/main/java/org/mycore/oai/set/MCROAISolrSetConfiguration.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-oai/src/main/java/org/mycore/oai/set/MCROAISolrSetConfiguration.java]: changed as suggested

**Review by toKrause — DISMISSED** [review]

**Comment by sebhofmann** [comment]
The .init method call is gone: https://github.com/MyCoRe-Org/mycore/commit/ec5de0a8ec4faf998b7e28ad2ddc7d4f6b7ef36a#diff-63774fe31cf1f8d417b64d01680d7715211199cbb259f3e5699188a4c040b81eL51

**Review by kkrebs — COMMENTED** [review]
works now with current MIR main

**Comment by rsteph-de** [comment]
the mix of calls to constructor and init() is a little bit confusing

**Review by toKrause — APPROVED** [review]

**Pull request merged**